### ISSUE TITLE
Paging of Row element is confusing and badly documented

### ISSUE THREAD

**Issue by SvizelPritula** [QuestPDF/QuestPDF#66]
The paging of a Row element works relatively intuitively as long as exactly one Columnt overflows to the next page. In that case that Column gets split between multiple pages and all other Columns get duplicated.

Weirdness arises, however, when multiple Columns overflow. In that case all of them get repeated until all happen to end on the same page. This easly makes the document very long. For example, the following code:

```
using QuestPDF.Drawing;
using QuestPDF.Fluent;
using QuestPDF.Helpers;
using QuestPDF.Infrastructure;

namespace Multicolumn;

public class Document : IDocument
{
    public DocumentMetadata GetMetadata() => DocumentMetadata.Default;

    public void Compose(IDocumentContainer container)
    {
        container
            .Page(page =>
            {
                page.Margin(50);
                page.Size(PageSizes.A4);

                page.Content().Row(row =>
                {
                    row.RelativeColumn().Element(c => CreateBoxes(c, 5));
                    row.RelativeColumn().Element(c => CreateBoxes(c, 6));
                    row.RelativeColumn().Element(c => CreateBoxes(c, 7));
                });
            });
    }

    void CreateBoxes(IContainer container, int count)
    {
        container.Stack(s =>
        {
            for (int i = 0; i < count; i++)
            {
                s.Item().Box()
                    .ExtendHorizontal().ExtendVertical()
                    .Background(Colors.Grey.Lighten3).Border(5)
                    .AlignCenter().AlignMiddle()
                    .Text(i);
            }
        });
    }
}
```

generates a 210 page PDF.

```
using QuestPDF.Drawing;
using QuestPDF.Fluent;
using QuestPDF.Helpers;
using QuestPDF.Infrastructure;

namespace Multicolumn;

public class Document : IDocument
{
    public DocumentMetadata GetMetadata() => DocumentMetadata.Default;

    public void Compose(IDocumentContainer container)
    {
        container
            .Page(page =>
            {
                page.Margin(50);
                page.Size(PageSizes.A4);

                page.Content().Row(row =>
                {
                    row.RelativeColumn().Element(c =>
                    {
                        c.Stack(s =>
                        {
                            CreateBox(s.Item().ShowOnce(), "X");

                            CreateBoxes(s.Item(), 2);
                        });
                    });
                    row.RelativeColumn().Element(c => CreateBoxes(c, 2));
                });
            });
    }

    void CreateBoxes(IContainer container, int count)
    {
        container.Stack(s =>
        {
            for (int i = 0; i < count; i++)
            {
                CreateBox(s.Item(), i.ToString());
            }
        });
    }

    void CreateBox(IContainer container, string text)
    {
        container.Box()
            .ExtendHorizontal().ExtendVertical()
            .Background(Colors.Grey.Lighten3).Border(5)
            .AlignCenter().AlignMiddle()
            .Text(text);
    }
}
```

generates an infinite PDF.

Most of this can be avoided with ShowOnce, but I think it should be mentioned in the documentation.
I also think it might not be a bad idea not to repeat any column that overflows itself.

**Comment by SvizelPritula** [comment]
Alternatively, it might be more intuitive to remove the text repeating behaviour altogether and create a horizontal decoration element.

**Comment by MarcinZiabek** [comment]
Thank you for finding this issue and providing suggestions. 

There is a reason why the Row element behaves like this. Sometimes you just want to repeat the column's content on every single page. This is one of the reasons why the ShowOnce element was introduced. So not to limit developers but to give choice. 

You may be right. This behaviour should be better documented. Or, we should change the default behaviour and give the per-column flag. Indeed, although repeating content is useful - I bet this is not the usual requirement and expectation.

However, there is something suspicious regarding generating over 200 pages - something good to look at :)

**Comment by MarcinZiabek** [comment]
Thank you for finding this issue! It is fixed in the QuestPDF 2022.02 release 😁 Closing
